src/components/views/avatars/MemberAvatar.tsx

```diff
@@ -46,100 +47,85 @@ interface IProps extends Omit<React.ComponentProps<typeof BaseAvatar>, "name" |
     hideTitle?: boolean;
 }
 
-interface IState {
-    name: string;
-    title: string;
-    imageUrl?: string;
-}
-
-export default class MemberAvatar extends React.PureComponent<IProps, IState> {
-    public static defaultProps = {
-        width: 40,
-        height: 40,
-        resizeMethod: 'crop',
-        viewUserOnClick: false,
-    };
-
-    constructor(props: IProps) {
-        super(props);
-
-        this.state = MemberAvatar.getState(props);
-    }
+export default function MemberAvatar({
+    width,
+    height,
+    resizeMethod = 'crop',
+    viewUserOnClick,
+    ...props
+}: IProps) {
+    const cli = useContext(MatrixClientContext);
+    const card = useContext(CardContext);
+    const roomContext = useContext(RoomContext);
+
+    const [name, setName] = useState<string | undefined>();
+    const [title, setTitle] = useState<string | undefined>();
+    const [imageUrl, setImageUrl] = useState<string | undefined>();
+    const [userId, setUserId] = useState<string | undefined>();
+
+    useEffect(() => {
+        let member = props.member;
 
-    public static getDerivedStateFromProps(nextProps: IProps): IState {
-        return MemberAvatar.getState(nextProps);
-    }
+        const useOnlyCurrentProfiles = (member && !props.forceHistorical
+            && SettingsStore.getValue("useOnlyCurrentProfiles"))
+            || roomContext?.timelineRenderingType === TimelineRenderingType.ThreadsList
+            || roomContext?.timelineRenderingType === TimelineRenderingType.Thread;
 
-    private static getState(props: IProps): IState {
-        let member = props.member;
-        if (member && !props.forceHistorical && SettingsStore.getValue("useOnlyCurrentProfiles")) {
-            const room = MatrixClientPeg.get().getRoom(member.roomId);
+        if (useOnlyCurrentProfiles) {
```

Comment:
This code is duplicated in the component below, maybe we can split this hook in multiple hook and derived state:

```ts
function getMember(...) {
   if (SettingsStore.getValue("useOnlyCurrentProfiles")
        || roomContext.timelineRenderingType === TimelineRenderingType.ThreadsList
        || roomContext.timelineRenderingType === TimelineRenderingType.Thread
    ) {
        const room = cli.getRoom(mxEvent.getRoomId());
        if (room) {
            member = room.getMember(mxEvent.getSender());
        }
    return member
}

function MemberAvatar() {
   const member = props.member && !props.forceHistorical && getMember()
   // maybe use memo to only compute when needed
   const name = member?.name || props.fallbackUserId || ''
  // maybe use memo to only compute when needed
   const userTitle = UserIdentifierCustomisations.getDisplayUserIdentifier(
                member.userId, { roomId: member?.roomId },
            );
   // maybe use memo to only compute when needed
   const title = member?.name ? props.title || userTitle :  props.fallbackUserId || ''
   const userId = member?.userId ?? props.fallbackUserId
   // maybe use memo to only compute when needed
   const imageUrl = member?.name && member.getMxcAvatarUrl() && mediaFromMxc(...)

   ...
}
```